### PR TITLE
[FLOC-2032] Introduce ClusterHQ Labs documentation

### PR DESCRIPTION
**Pull request by lukemarsden**
This is a set of documentation for projects performed by ClusterHQ Labs.

As agreed with engineering, these docs do not need to be reviewed for content, but merely for issues that may cause problems with the build. They have already been internally reviewed by us nice labs folks.

However if you do spot any restructuredText issues or typos etc, flagging them up before merging would be much appreciated!

### CODE REVIEW

**Comment by adamtheturtle** [comment]
The following is a full review, as per https://zulip.com/#narrow/stream/engineering/subject/pragmatism.20in.20reviews. However I won't make a judgement of which review comments must be addressed now.

File: docs/labs/compose.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+==================
+
+To make use of Flocker volumes with Compose, you will need to specify a ``volume_driver`` field in your ``docker-compose.yml`` file.
+
+Here is an example of a simple application that has 2 containers ``web`` and ``redis``.  Notice how the ``redis`` container has a data volume and has the ``volume_driver`` field set to ``flocker``.
+
+.. code-block:: yaml
+
+    web:
+      image: binocarlos/moby-counter
```
Comment (adamtheturtle) [inline docs/labs/compose.rst]: Having things in our docs tied to a particular employee is setting ourselves up for failure (e.g. if that employee leaves).

File: docs/labs/compose.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+First, you need to :ref:`install Flocker <labs-installer>` and the :ref:`Flocker Docker plugin <labs-docker-plugin>`.
+You can use our experimental  :ref:`Flocker Installer <labs-installer>` to do this.
+
+Then, you need a version of Compose that supports Flocker volumes.
+
+Run the following command to do this (this branch that will be installed corresponds to `this pull request <https://github.com/docker/compose/pull/1502>`_):
+
+.. prompt:: bash $
+
+    sudo pip install git+https://github.com/lukemarsden/compose.git@volume_driver
```
Comment (adamtheturtle) [inline docs/labs/compose.rst]: oh noes `sudo pip`.

File: docs/labs/docker-plugin.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+Flocker Docker plugin
+=====================
+
+The Flocker Docker plugin is a `Docker volumes plugin <https://github.com/docker/docker/blob/master/experimental/plugins_volume.md>`_, connecting Docker on a host directly to Flocker, where Flocker agents will be running on the same host and hooked up to the Flocker control service.
+
+This diagram explains how the architecture of a Flocker cluster with the Docker plugin would look if the user is also using :ref:`Docker Swarm <labs-swarm>` and :ref:`Docker Compose <labs-compose>`:
+
+.. The source file for this diagram is in Engineering/Labs folder on GDrive: https://drive.google.com/open?id=0B3gop2KayxkVc1g3R1AyQzFNODQ
+
+.. image:: docker-plugin-platform-architecture.png
```
Comment (adamtheturtle) [inline docs/labs/docker-plugin.rst]: Source files in source control please. Drive is not source control.

File: docs/labs/installer.rst [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+This will configure API certificates for the Flocker Docker plugin and push them to your nodes - it will name them ``/etc/flocker/plugin.{crt,key}`` on the nodes.
+
+It will install the Flocker Docker plugin, and write a service file (``upstart``/``systemd``) for the plugin (as described in the :ref:`manual installation instructions for the Flocker Docker plugin <labs-docker-plugin>`.
+
+It will also download and install an experimental Docker binary that supports the ``--volume-driver`` flag and restart the Docker service.
+
+It supports several optional environment variables:
+
+* ``DOCKER_BINARY_URL`` - the URL to download a customized Docker binary from
```
Comment (adamtheturtle) [inline docs/labs/installer.rst]: Use `:envvar:`.

File: docs/labs/docker-plugin.rst [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+* If the volume exists on the current host, the container is started straight away.
+
+Multiple containers can use the same Flocker volume (by referencing the same volume name, or by using Docker's ``--volumes-from``) so long as they are running on the same host.
+
+Demo
+====
+
+This demo shows both the Flocker Docker plugin in conjunction with the :ref:`Volumes CLI <labs-volumes-cli>` and :ref:`Volumes GUI <labs-volumes-gui>`.
+
+.. raw:: html
```
Comment (adamtheturtle) [inline docs/labs/docker-plugin.rst]: Use https://pypi.python.org/pypi/sphinxcontrib.youtube

File: docs/labs/docker-plugin.rst [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+Manual Installation on Ubuntu 14.04
+===================================
+
+First install Flocker on some hosts.
+These instructions assume you have followed the :ref:`official Flocker install instructions <installflocker>`.
+See the :ref:`unofficial installer <labs-installer>` for a quicker method.
+
+On the same machine where you ran ``flocker-ca`` while installing Flocker, :ref:`generate a new API user certificate and key <generate-api>` for a user named ``plugin``.
+Upload these files to ``/etc/flocker/plugin.key`` and ``/etc/flocker/plugin.crt`` on the hosts where you want to run the Flocker Docker plugin.
```
Comment (adamtheturtle) [inline docs/labs/docker-plugin.rst]: Files should use ``:file:``.

File: docs/labs/volumes-gui.rst [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    Refer to the instructions in the :ref:`experimental installer <labs-installer>`.
+
+Step 2 - load up the experimental Flocker GUI
+=============================================
+
+Go to `http://localhost/client/#/nodes/list <http://localhost/client/#/nodes/list>`_ or, if you are using ``boot2docker``:
+
+.. prompt:: bash $
+
+    open http://$(boot2docker ip)/client/#/nodes/list
```
Comment (adamtheturtle) [inline docs/labs/volumes-gui.rst]: Does this work on all supported platforms? I know it works on OS X.

File: docs/labs/volumes-gui.rst [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+Step 3
+======
+
+There is no step 3.
```
Comment (adamtheturtle) [inline docs/labs/volumes-gui.rst]: Weird

File: docs/introduction/index.rst [hunk 9]
```diff
@@ -42,12 +42,14 @@ Flocker supports block-based shared storage such as Amazon EBS, Rackspace Cloud
...
 
+.. _flocker-containers-architecture:
+
 .. image:: images/flocker-architecture.svg
    :alt: Flocker architecture with shared storage backend.
          The Flocker Control Service and containers hosts can run on a VM or bare metal servers.
 		 The Flocker Agent running on each host speaks to the shared storage backend to create and moutn volumes to individual containers.
 
-Flocker also has planned integrations with major orchestration tools such as Docker Swarm, Kubernetes and Apache Mesos. More information on this integration is coming soon.
+Flocker also has planned integrations with major orchestration tools such as Docker Swarm, Kubernetes and Apache Mesos. More information on this integration is :ref:`available in the Labs section <labs-projects>`.
```
Comment (adamtheturtle) [inline docs/introduction/index.rst]: :question: Links to other projects

File: docs/labs/compose.rst [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+docker-compose.yml
+==================
+
+To make use of Flocker volumes with Compose, you will need to specify a ``volume_driver`` field in your ``docker-compose.yml`` file.
```
Comment (adamtheturtle) [inline docs/labs/compose.rst]: Makes it seem like we call it "docker-compose.yml" but we don't anywhere in the docs it seems, we call it the deployment configuration file.

**Comment by lukemarsden** [comment]
Closing this PR so that we can do a rebase against `release/flocker-1.0.0` (with @wallrj).

**Pull request closed without merging**